### PR TITLE
INTG-456 API documentation updated for the workflows

### PR DESCRIPTION
**Pull request by mrinalirao**
API documentation updated for the workflow endpoints

### CODE REVIEW

**Comment by dimitrikoubaroulis** [comment]
@timm088 can you please review this as part of your taking ownership of the dev portal. Will review too.

File: source/index.md [hunk 1]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
 If the <code>total</code> is greater than the <code>count</code>, more templates can be retrieved by either increasing the
 <code>limit</code>, or repeating the search with a new <code>modified_after</code> value.
 </aside>
+
+# Workflows
+
+This allows you to enable real-time integrations with our API's.
```
Comment (dimitrikoubaroulis) [inline source/index.md]: This allows you to receive near real-time audit event notifications from SafetyCulture.

File: source/index.md [hunk 2]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+Workflows can be registered for an event type (for example: audit_completed), so that a notification will be sent to you when this event takes place.
```
Comment (dimitrikoubaroulis) [inline source/index.md]: will be sent to you in the form of a POST HTTP request when this event takes place.

File: source/index.md [hunk 3]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+          "audit_completed", "audit_archived"
+      ],
+      "steps": [
+          {
+              "url": "https://example.com",
+              "type": "webhook",
+              "id": "workflowstep_63aa1e21816f4bf78e6d3a4f7cbfaef1"
+          }
+      ],
+      "user_id": "user_7db5002c72754d2a99ac1e4b82f088be",
```
Comment (dimitrikoubaroulis) [inline source/index.md]: Are these real IDs from our production system? Make them fake please if that's the case. If from sandpit or local that's fine.

File: source/index.md [hunk 4]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+      "org_id": "role_7db5002c72754d2a99ac1e4b82f088be",
+      "enabled": true,
+      "updated_at": "2017-06-19T03:51:12.299Z",
+      "created_at": "2017-06-19T03:51:12.299Z",
+      "version": "1.0.0"
+}
+```
+This endpoint allows you to register a workflow for a given event type.
+You can only register for the following event types:
```
Comment (dimitrikoubaroulis) [inline source/index.md]: The following event types are supported so far:

File: source/index.md [hunk 5]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+
+### Request Body
+
+The body should use the content type `application/json` with the following required parameters in the
+body:
+
+Parameter          | Description
+---------          | -----------
+`trigger_events`   | The audit events for which you would want to be notified. Valid values are `audit_completed`, `audit_archived`, `audit_created`, `audit_updated` and `audit_unarchived`
+`steps`            | Contains th workflow URL to be registered and the type of the workflow. As of now only the type `webhook` is supported
```
Comment (dimitrikoubaroulis) [inline source/index.md]: th => the

File: source/index.md [hunk 6]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+
+### Response
+
+The response will have the content type `application/json` and include the following parameters:
+
+Parameter          | Description
+---------          | -----------
+`workflow_id`      | The ID of the workflow
+`trigger_events`   | The audit events for which you would want to be notified. Valid values are `audit_completed`, `audit_archived`, `audit_created`, `audit_updated` and `audit_unarchived`
+`steps`            | Contains th workflow URL that is registered and the type of the workflow. As of now only the type `webhook` is supported. It also contains the `id` of the step
```
Comment (dimitrikoubaroulis) [inline source/index.md]: th => the

File: source/index.md [hunk 7]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+`user_id`          | User ID of the user who registered the workflow
+`org_id`           | Organization ID of the user who registered the workflow
+`enabled`          | Flag to check if workflow needs to be triggered. Defaults to `true`
```
Comment (dimitrikoubaroulis) [inline source/index.md]: If 'false' no events will be triggered from this workflow. Defaults to `true

File: source/index.md [hunk 8]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+
+The response will have the content type `application/json` and include the following parameters:
+
+Parameter          | Description
+---------          | -----------
+`workflow_id`      | The ID of the workflow
+`trigger_events`   | The audit events for which you would want to be notified. Valid values are `audit_completed`, `audit_archived`, `audit_created`, `audit_updated` and `audit_unarchived`
+`steps`            | Contains th workflow URL that is registered and the type of the workflow. As of now only the type `webhook` is supported. It also contains the `id` of the step
+`user_id`          | User ID of the user who registered the workflow
+`org_id`           | Organization ID of the user who registered the workflow
```
Comment (dimitrikoubaroulis) [inline source/index.md]: We should cover that a user can only register workflows for the whole of their organisation and they need to have the `Manage Role` permission on that organisation to be able to do so. Not sure if this is called 'Manage Role' in the UI, probably called something else. Let's find out please and update.

File: source/index.md [hunk 9]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+
+### HTTP Request
+
+`GET https://api.safetyculture.io/workflows/<workflow_id>`
+
+### URL Parameters
+
+Parameter    | Description
+---------    | -----------
+workflow ID  | The ID of the workflow setting object to return
```
Comment (dimitrikoubaroulis) [inline source/index.md]: I'd remove 'setting object'

File: source/index.md [hunk 10]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+
+### Response
+
+The response will have the content type `application/json` and include the following parameters:
+
+Parameter          | Description
+---------          | -----------
+`workflow_id`      | The ID of the workflow
+`trigger_events`   | The audit events for which you would want to be notified. Valid values are `audit_completed`, `audit_archived`, `audit_created`, `audit_updated` and `audit_unarchived`
+`steps`            | Contains th workflow URL that is registered and the type of the workflow. As of now only the type `webhook` is supported. It also contains the `id` of the step
```
Comment (dimitrikoubaroulis) [inline source/index.md]: th => the (please check all places)

File: source/index.md [hunk 11]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+`user_id`          | User ID of the user who registered the workflow
+`org_id`           | Organization ID of the user who registered the workflow
+`enabled`          | Flag to check if workflow needs to be triggered. Defaults to `true`
```
Comment (dimitrikoubaroulis) [inline source/index.md]: Please update all places with updated wording from above

File: source/index.md [hunk 12]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+`updated_at`       | Last modified timestamp of this workflow setting
+`created_at`       | Creation timestamp of this workflow setting
+`version`          | Version of this workflow setting object
+
+Status code     | Description
+-----------     | -----------
+200 OK          | Retrieve successful
+400 Bad Request | Invalid request
+403 Forbidden   | The user does not have permission to register a workflow
```
Comment (dimitrikoubaroulis) [inline source/index.md]: This is not about registering here

File: source/index.md [hunk 13]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+
+### Request Body
+
+The body should use the content type `application/json` with the following required parameters in the
+body:
+
+Parameter          | Description
+---------          | -----------
+`trigger_events`   | The audit events for which you would want to be notified. Valid values are `audit_completed`, `audit_archived`, `audit_created`, `audit_updated` and `audit_unarchived`
+`steps`            | Contains th workflow URL to be registered and the type of the workflow. As of now only the type `webhook` is supported
```
Comment (dimitrikoubaroulis) [inline source/index.md]: th => the

File: source/index.md [hunk 14]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+
+Note: You can specify only the field that needs to be updated
```
Comment (dimitrikoubaroulis) [inline source/index.md]: fields that need to be updated i.e. do a partial update

File: source/index.md [hunk 15]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+`enabled`          | Flag to check if workflow needs to be triggered. Defaults to `true`
+`updated_at`       | Last modified timestamp of this workflow setting
+`created_at`       | Creation timestamp of this workflow setting
+`version`          | Version of this workflow setting object
+
+Status code     | Description
+-----------     | -----------
+200 OK          | Updation successful
+400 Bad Request | Invalid request or payload
+403 Forbidden   | The user does not have permission to register a workflow
```
Comment (dimitrikoubaroulis) [inline source/index.md]: to update

File: source/index.md [hunk 16]
```diff
@@ -1434,3 +1434,348 @@ Parameter      | Description
...
+
+### Response
+
+The successful response for this request will be {"ok": true}
+
+Status code     | Description
+-----------     | -----------
+200 OK          | Deletion successful
+400 Bad Request | Invalid request 
+403 Forbidden   | The user does not have permission to access the audit with audit ID
```
Comment (dimitrikoubaroulis) [inline source/index.md]: fix

File: source/index.md (hunk 3 above)
Comment (mrinalirao) [inline source/index.md]: they are from my local

File: source/index.md [hunk 17]
```diff
@@ -1434,3 +1434,355 @@ Parameter      | Description
...
+
+### Request Body
+
+The body should use the content type `application/json` with the following required parameters in the
+body:
+
+Parameter          | Description
+---------          | -----------
+`trigger_events`   | The audit events for which you would want to be notified. Valid values are `audit_completed`, `audit_archived`, `audit_created`, `audit_updated` and `audit_unarchived`
+`steps`            | Contains the workflow URL to be registered and the type of the workflow. As of now only the type `webhook` is supported
```
Comment (koresar) [inline source/index.md]: Better say: "Array of workflow steps. See table below of available workflow step types."

Then create table below having two columns and only one row:
> `webhook` | Must contain properties `"type": "webhook"` and `"url"`. Only the `HTTPS` URL schema is allowed.

**Comment by koresar** [comment]
LGTM

**Pull request merged**